### PR TITLE
MB-1584 ServiceAreaDest lookup

### PR DESCRIPTION
**Pull request by garrettqmartin8**
## Description

We have a lookup to find the service area origin when creating a payment request - we now need a way to lookup the service area destination.

## Setup

- `make db_dev_e2e_populate`
- Create a file called `pr_payload.json` in the root of the project:

```
{
  "body": {
    "isFinal": false,
    "moveTaskOrderID": "9c7b255c-2981-4bf8-839f-61c7458e2b4d",
    "serviceItems": [
      {
        "id": "94bc8b44-fefe-469f-83a0-39b1e31116fb" // service item for Dom. Destination Price, which uses the lookup
      }
    ]
  }
}
```
- Run `go run ./cmd/prime-api-client --insecure create-payment-request --filename pr_payload.json -v | jq`
- Verify that the response in your terminal includes this as part of the params:

```
{
  "eTag": "MjAyMC0wNy0xMFQxODowMToyNC4xMDk2MzNa",
   "id": "cd3a95d7-6d0e-43e7-913f-61c3978dfb14",
   "key": "ServiceAreaDest",
   "origin": "SYSTEM",
   "paymentServiceItemID": "3540b180-ae7f-4f4b-b41d-95cb7e05004f",
   "type": "STRING",
   "value": "080"
}
```

## Code Review Verification Steps

* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-1584) for this change

### CODE REVIEW

File: pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+	zip := mtoServiceItem.MTOShipment.DestinationAddress.PostalCode
+	zip3 := zip[0:3]
+
+	var domesticServiceArea models.ReDomesticServiceArea
+
+	query := db.Q().
+		Join("re_zip3s", "re_zip3s.domestic_service_area_id = re_domestic_service_areas.id").
+		Join("re_contracts", "re_contracts.id = re_domestic_service_areas.contract_id").
+		Where("zip3 = ?", zip3)
```
Comment (pearl-truss) [inline pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go:55]: @garrettqmartin8 does this need to use the DefaultContractCode, like with the ServiceAreaOriginLookup?

Comment (garrettqmartin8) [reply pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go:55]: Yes it does, good catch @pearl-truss! Will update now.

Comment (pearl-truss) [reply pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go]: Perfect, it looks good now!

Comment (pearl-truss) [reply pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go]: Oh wait, one of the server test is failing

File: pkg/testdatagen/scenario/e2ebasic.go [hunk 2]
```diff
@@ -1209,15 +1209,29 @@ func (e e2eBasicScenario) Run(db *pop.Connection, userUploader *uploader.UserUpl
...
+	estimated := unit.Pound(1400)
 	mtoShipment4 := testdatagen.MakeMTOShipment(db, testdatagen.Assertions{
 		MTOShipment: models.MTOShipment{
-			ID:                  uuid.FromStringOrNil("c3a9e368-188b-4828-a64a-204da9b988c2"),
-			RequestedPickupDate: swag.Time(time.Now()),
-			ScheduledPickupDate: swag.Time(time.Now().AddDate(0, 0, -1)),
+			ID:                   uuid.FromStringOrNil("c3a9e368-188b-4828-a64a-204da9b988c2"),
+			RequestedPickupDate:  swag.Time(time.Now()),
+			ScheduledPickupDate:  swag.Time(time.Now().AddDate(0, 0, -1)),
+			PrimeEstimatedWeight: &estimated, // so we can price Dom. Destination Price
```
Comment (gidjin) [inline pkg/testdatagen/scenario/e2ebasic.go:1218]: Do we need `PrimeActualWeight` too?

File: pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go (hunk 1 above)
Comment (garrettqmartin8) [reply pkg/payment_request/service_param_value_lookups/service_area_dest_lookup.go]: Seems to be a hiccup in Circle - tests are passing now.

File: pkg/testdatagen/scenario/e2ebasic.go (hunk 2 above)
Comment (garrettqmartin8) [reply pkg/testdatagen/scenario/e2ebasic.go:1218]: Yeah it looks like we need `PrimeActualWeight`, but it is already being set. In my dev db, all `mto_shipments` created in `e2e_basic.go` have the field set.

**Pull request merged**